wasm/main.go

```diff
@@ -196,7 +204,274 @@ func parseJWTCredential() js.Func {
 		json.Unmarshal(credBytes, &resultObj)
 
 		return js.ValueOf(resultObj)
+	})
+}
+
+// verifyJWTCredential
+//
+// @Summary Verify JWT Verifiable Credential
+// @Description Verify a JWT Verifiable Credential by checking its signature and issuer against the provided DID
+// @Param jwtCredential string "The JWT string representing the Verifiable Credential"
+// @Param didKey string "DID string to be used for verification"
+// @Success js.Value "true" "If the JWT Verifiable Credential is successfully verified"
+// @Error js.Value "An error object with a message describing the error"
+func verifyJWTCredential() js.Func {
+	return js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		if len(args) != 2 {
+			return generateError(errors.New("invalid arg count, need jwt credential string as argument"))
+		}
+		keyType := crypto.Ed25519
+
+		vcJWT := args[0].String()
+		didKey := args[1].String()
+
+		didDoc, err := did.DIDKey(didKey).Expand()
+		if err != nil {
+			return generateError(err)
+		}
+
+		publicKeyBytes, err := base58.Decode(didDoc.VerificationMethod[0].PublicKeyBase58)
+		if err != nil {
+			return generateError(err)
+		}
+
+		cryptoPubKey, err := crypto.BytesToPubKey(publicKeyBytes, keyType)
+		if err != nil {
+			return generateError(err)
+		}
+
+		cred, err := signing.ParseVerifiableCredentialFromJWT(vcJWT)
+		if err != nil {
+			return generateError(err)
+		}
+
+		// TODO: Get kid from vcJWT?
+		verifierKid := cred.Issuer.(string)
+
+		jwtVerifier, err := crypto.NewJWTVerifier(verifierKid, cryptoPubKey)
+		if err != nil {
+			return generateError(err)
+		}
+
+		_, err = signing.VerifyVerifiableCredentialJWT(*jwtVerifier, vcJWT)
```

Comment:
it would be great to uptake this commit of the ssi sdk which fixes a few issues around generating and signing jwts https://github.com/TBD54566975/ssi-sdk/commit/cbd9818e83671142cad34430c20e5a59eaea44cf